### DISCUSSION TITLE
Pyodide build package with enabling memory growth, and allow the program to allocate all the way up to 4GB of memory

### DISCUSSION THREAD

**Discussion by airen1986** [pyodide/pyodide#5140] [Q&A] [answered]
Hi Team,

How can build a pyodide package with custom flags like:

``` emcc -s ALLOW_MEMORY_GROWTH -s MAXIMUM_MEMORY=4GB```

Thanks

**Comment by hoodmane** [comment]
These are flags that are passed when linking the Python interpreter itself. We pass `-s ALLOW_MEMORY_GROWTH`:
https://github.com/pyodide/pyodide/blob/main/Makefile.envs#L155
If you want to change these flags and link your own Python interpreter, you can modify `Makefile.envs` and follow [the instructions for building from source](https://pyodide.org/en/stable/development/building-from-sources.html)

**Reply by airen1986** [reply]
I have build a python package where I am hitting 2GB memory limit, however based on this blog: https://v8.dev/blog/4gb-wasm-memory , it can be extended to 4GB with the above flags. Question: Why is it not defaulted to 4GB in pyodide?

**Reply by hoodmane** [reply]
No particular reason. Assuming it also works in Firefox and Safari this sounds fine. PR welcome.

**Comment by dylanjcastillo** [comment, marked as answer]
For anyone wondering now, `MAXIMUM_MEMORY` is now set to 4GB by default: https://github.com/pyodide/pyodide/pull/5167
